### ISSUE TITLE
StackOverflowError while processing generic endpoint class

### ISSUE THREAD

**Issue by tiguchi** [stoicflame/enunciate#699]
**Enunciate version:** 2.9.1

Hello,

I stumbled into a `java.lang.StackOverflowError` caused by Enunciate while it is trying to analyze a generic JAX-RS endpoint class in my web API project.

I could pin-point the cause in a minimal example that reproduces the bug:

https://github.com/tiguchi/EnunciateStackOverflow

See the following file:

[CollectionResource.java, line 39](https://github.com/tiguchi/EnunciateStackOverflow/blob/master/src/main/java/test/CollectionResource.java#L39)

```java
	/**
	 * Gets endpoint for accessing a specific resource object by ID.
	 * <p>
	 * <b>CAUSES THE StackOverflowError EXCEPTION</b>
	 * 
	 * @param id
	 *            Resource object ID.
	 * @return Endpoint for resource.
	 */
	@Path("{id}")
	public Resource<T> getById(@PathParam("id") ID<T> id) {
		return asResource(id);
	}
```

The following line of Enunciate code gets stuck in a recursion: [TypeVariableContext.java, line 60](https://github.com/stoicflame/enunciate/blob/master/javac-support/src/main/java/com/webcohesion/enunciate/javac/decorations/type/TypeVariableContext.java#L60)

**Comment by stoicflame** [comment]
Thanks for the report and the reproduction effort. I'll get it fixed for 2.10.

**Comment by stoicflame** [comment]
[Enunciate 2.10 is now available](https://github.com/stoicflame/enunciate/releases/tag/v2.10.0).
